src/routes/account/+page.server.ts

```diff
@@ -0,0 +1,101 @@
+import type { Actions, PageServerLoad } from './$types';
+import { error, redirect } from '@sveltejs/kit';
+import {
+  getStripeCustomerWithSubscriptions,
+  getBlendProSubscription,
+  getCustomerPortalSession,
+  createStripeSession,
+  isSubscribedToBlendPro,
+  isOrganizationMember,
+} from '$lib/server/subscriptionUtils';
+import { checkSessionAuth, getUserData, getUserOrganizations, isUserGlobalAdmin, readPath, writePath } from '$lib/server/firebaseUtils';
+
+export const load = (async ({ url, cookies }) => {
+  const actionParam = url.searchParams.get('action') ?? '';
+  const uid = (await checkSessionAuth(cookies, {
+    loginRedirect: `account${url.search}`,
+  })).uid;
+
+  const userData = await getUserData(uid);
+  const email = userData.email!;
+  const name = userData.displayName ?? 'Blend User';
+  const isGlobalAdmin = isUserGlobalAdmin(uid); 
+  const orgIds = await getUserOrganizations(uid);
+  const organizations = await Promise.all(
+    orgIds.map(async (orgId) => {
+      const organization = await readPath<Database.Organization>(`/organizations/${orgId}`);
+      return {
+        id: orgId, 
+        name: organization?.public.name,
+        role: (await isGlobalAdmin) ? 'admin' : organization?.private?.members?.[uid]?.role,
+      };
+    }),
+  );
+  const customer = await getStripeCustomerWithSubscriptions(uid);
+
+  // Redirect to Stripe Checkout if necessary
+  if (actionParam && !isSubscribedToBlendPro(customer) && !(await isOrganizationMember(uid))) {
+    switch (actionParam) {
+      case 'upgrade': {
+        const stripeSession = await createStripeSession(uid, email, name, url.origin);
+        throw redirect(303, stripeSession.url!);
+      }
+      case 'choosePlan':
+        throw redirect(303, '/account/plan');
```

Comment:
I believe this case is unreachable because of my other comment. We can either remove it and live with the behavior, or fix the other area somehow